### ISSUE TITLE
Array size expression in type alias does not recognize scope

### ISSUE THREAD

**Issue by adlerd** [rust-analyzer/rust-analyzer#5636]
In the array size expression part of a type alias, rust-analyzer doesn't recognize the name of an adjacent const item as being in scope. It offers an action to import the name. Doing this once imports the bare name (which is not legal); doing it again causes an ERROR report.

Example:
```rust
pub const SIZE: usize = 42;
pub type ROADS = [(); SIZE /* <- token of interest */ ];
```

The marked token does not get the expected semantic highlight and hovers produce no info. Renaming the const does not update this token. (However, `goto definition` does work...) There is an action to `` Import `SIZE` ``. Activating that action produces this document, which isn't valid rust due to a name being redefined, and the token still isn't recognized:

```rust
use SIZE;

pub const SIZE: usize = 42;
pub type ROADS = [(); SIZE];
```

Doing the action again produces this output:
```
ERROR [7/31/2020, 10:32:42 PM]: Assertion failed: bad ws edit: []
INFO [7/31/2020, 10:32:46 PM]: Extension version: 0.3.261-nightly
```

Variants: This also occurs with a const fn instead of a const usize. Using an explicit path to an item in a child module also doesn't link the names, but the first time you use the `import` action, it makes the expected update to the document; a second `import` tries to import the bare name and a third gives the error.

**Comment by flodiebold** [comment]
I don't get the code action, @adlerd is this still happening for you?

**Comment by adlerd** [comment]
This appears to be fixed now in 0.2.424.
